gnucash/gnucash.cpp

```diff
@@ -155,7 +161,7 @@ scm_run_gnucash (void *data, [[maybe_unused]] int argc, [[maybe_unused]] char **
     /* Load the scm config files before starting up the gui. This ensures that
      * custom reports have been read into memory before the Reports
      * menu is created. */
-    Gnucash::gnc_load_scm_config();
+    Gnucash::gnc_load_scm_config (gui_update_message_cb);
```

Comment:
I had in mind that you'd do
```
     Gnucash::gnc_load_scm_config(
                               [](const gchar *msg)
                               {
                                   gnc_update_splash_screen (msg, GNC_SPLASH_PERCENTAGE_UNKNOWN);
                                   PINFO ("%s", msg);
                                });
```
But using the named lambda is OK too, Just move the decls inside their respective function scopes. Right now they're global variables, not a Good Thing.

Reply:
the cb is needed in several functions in `gnucash-commands.cpp`

Reply:
It's also a one-liner that's mostly a no-op, so it wouldn't hurt anything to use the immediate lambda in all three places. It might even be better: By creating the lambda in each function it might capture enough context to print the right function in the log message. I'm not sure about that, but it might.

OTOH it doesn't matter if it's a lambda or a regular function, and `std::function` works with both. Either way declare it `static` so it's only file-global and not global-global.

Reply:
Agree that inlining the single liner is better. A top-level lambda shows 

```
* 19:57:07  WARN <gnc.gui> [<lambda()] Loading system wide Guile extensions…
* 19:57:07  WARN <gnc.gui> [<lambda()] Loading user specific Guile extensions…
```

and the inlined lambda shows
```
* 19:57:41  WARN <gnc.gui> [scm_report_list()] Loading system wide Guile extensions…
* 19:57:41  WARN <gnc.gui> [scm_report_list()] Loading user specific Guile extensions…
```